chainercv/links/model/resnet/resblock.py

```diff
@@ -0,0 +1,107 @@
+import chainer
+import chainer.functions as F
+
+from chainercv.links import Conv2DBNActiv
+
+
+class ResBlock(chainer.Chain):
+
+    """A building block for ResNets.
+
+    in --> Bottleneck with shortcut --> Bottleneck * (n_layer - 1) --> out
+
+    Args:
+        n_layer (int): The number of layers used in the building block.
+        in_channels (int): The number of channels of input arrays.
+        mid_channels (int): The number of channels of intermediate arrays.
+        out_channels (int): The number of channels of output arrays.
+        stride (int or tuple of ints): Stride of filter application.
+        initialW (4-D array): Initial weight value used in
+            the convolutional layers.
+        stride_first (bool): This determines the behavior of the
+            bottleneck with a shortcut. If :obj:`True`, apply strided
+            convolution with the first convolution layer.
+            Otherwise, apply strided convolution with the
+            second convolution layer.
+
+    """
+
+    def __init__(self, n_layer, in_channels, mid_channels,
+                 out_channels, stride, initialW=None, stride_first=False):
+        super(ResBlock, self).__init__()
+        with self.init_scope():
+            self.a = Bottleneck(
+                in_channels, mid_channels, out_channels, stride,
+                initialW, residual_conv=True, stride_first=stride_first)
+            self._forward = ['a']
+            for i in range(n_layer - 1):
+                name = 'b{}'.format(i + 1)
+                bottleneck = Bottleneck(
+                    out_channels, mid_channels, out_channels, stride=1,
+                    initialW=initialW, residual_conv=False)
+                self.add_link(name, bottleneck)
+                self._forward.append(name)
+
+    def __call__(self, x):
+        for name in self._forward:
+            l = getattr(self, name)
+            x = l(x)
+        return x
+
+
+class Bottleneck(chainer.Chain):
+
+    """A bottleneck layer.
+
+    Args:
+        in_channels (int): The number of channels of input arrays.
```

Comment:
`input arrays` -> `(the) input array`? From my understanding, this link takes only one array.